CRM/Utils/Number.php

```diff
@@ -128,4 +128,24 @@ public static function formatLocaleNumeric(string $amount, $locale = NULL): stri
     return $formatter->format($amount);
   }
 
+  /**
+   * Parse a string into a floating-point number according to the current or supplied locale.
+   *
+   * @param string $amount
+   * @param string $locale
+   *
+   * @return float|bool
+   */
+  public static function parseLocaleNumeric(string $amount, string $locale = NULL) {
+    if ($amount === "") {
+      CRM_Core_Error::deprecatedWarning('Passing an empty string for amount is deprecated.');
+      return NULL;
+    }
+
+    $formatter = new \NumberFormatter($locale ?? CRM_Core_I18n::getLocale(), NumberFormatter::DECIMAL);
```

Comment:
So this is my biggest hesitation - it feels like 'yet another place with money formatting' 

We had been consolidating on the `Civi::format()` class with `machineMoney` being somewhat intended to fill this function but getting a bit distracted in the implementation. Or perhaps we decided we need that as well (hard to recall).

I think if you don't want to open up this whole format-function  can of worms you could just use the same `cleanMoney` that the money functions do - it doesn't really do anything very money-like to be honest  - but at least it's in use in lots of places already & as we deprecate `CRM_Core_Config::singleton()->monetaryThousandSeparator` it will get updated...